### PR TITLE
Ledger: refactor bazel packaging of tedger test tool

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
improve previous generalization from #8695

- use lf version instead keyword (like 'stable', 'latest', 'dev') to
  tag actual target.  This will allow two keywords to map to the same
  versions without doing twice the compilation/test work.

- use alias to map keywords tag target to versioned tag target.

- extend lookup logic to find test common dar.

- move package manage dar to test_commong.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json/BUILD.bazel [hunk 1]
```diff
@@ -260,7 +260,7 @@ da_scala_test(
     data = [
         ":Account.dar",
         "//docs:quickstart-model.dar",
-        "//ledger/test-common:model-tests.dar",
+        "//ledger/test-common:test-dars-stable",
```
Comment (remyhaemmerle-da) [inline ledger-service/http-json/BUILD.bazel:263]:
```suggestion
        "//ledger/test-common:model-tests-stable.dar",
```

File: language-support/java/codegen/src/ledger-tests/scala/com/digitalasset/TestUtil.scala [hunk 2]
```diff
@@ -33,12 +32,13 @@ import org.scalatest.Assertion
...
 import scala.language.implicitConversions
-
 import java.util.Arrays.asList
 
+import com.daml.bazeltools.BazelRunfiles
+
 object TestUtil {
 
-  def testDalf =
+  def testDalf: File =
```
Comment (remyhaemmerle-da) [inline language-support/java/codegen/src/ledger-tests/scala/com/digitalasset/TestUtil.scala:41]:
```suggestion
  def testDalf =
```

File: language-support/java/codegen/src/ledger-tests/scala/com/digitalasset/TestUtil.scala [hunk 3]
```diff
@@ -33,12 +32,13 @@ import org.scalatest.Assertion
 import scala.concurrent.Future
 import scala.jdk.CollectionConverters._
 import scala.language.implicitConversions
-
 import java.util.Arrays.asList
 
+import com.daml.bazeltools.BazelRunfiles
+
```
Comment (remyhaemmerle-da) [inline language-support/java/codegen/src/ledger-tests/scala/com/digitalasset/TestUtil.scala:38]:
```suggestion
```

File: rules_daml/daml.bzl [hunk 4]
```diff
@@ -245,14 +245,15 @@ def daml_compile(
...
+        project_name = None,
         **kwargs):
     "Build a DAML project, with a generated daml.yaml."
     if len(srcs) == 0:
         fail("daml_compile: Expected `srcs' to be non-empty.")
     daml_yaml = name + ".yaml"
     _daml_configure(
         name = name + ".configure",
-        project_name = name,
+        project_name = project_name or name,
```
Comment (remyhaemmerle-da) [inline rules_daml/daml.bzl:256]: :copyright: @cocreature

File: compiler/damlc/tests/BUILD.bazel [hunk 5]
```diff
@@ -8,6 +8,7 @@ load("//rules_daml:daml.bzl", "daml_compile")
 load("@build_environment//:configuration.bzl", "sdk_version")
 load("//daml-lf/language:daml-lf.bzl", "COMPILER_LF_VERSIONS", "lf_latest_version")
 load("//compiler/damlc:util.bzl", "ghc_pkg")
+load("//ledger/test-common:test-common.bzl", "test_common_configurations", "test_common_lf_targets")
```
Comment (cocreature) [inline compiler/damlc/tests/BUILD.bazel:11]: where are you using the new imports?

File: daml-script/dump/BUILD.bazel [hunk 6]
```diff
@@ -58,7 +58,7 @@ da_scala_test(
     data = [
         "//compiler/damlc",
         "//daml-script/daml:daml-script.dar",
-        "//ledger/test-common:dar-files",
+        "//ledger/test-common:test-dars-stable",
```
Comment (cocreature) [inline daml-script/dump/BUILD.bazel:61]: is it worth having an alias that defaults to the stable version?

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/PackageManagementServiceIT.scala [hunk 7]
```diff
@@ -16,11 +16,11 @@ import scala.concurrent.{ExecutionContext, Future}
 
 final class PackageManagementServiceIT extends LedgerTestSuite {
   private[this] val testPackageResourcePath =
-    "/ledger/ledger-api-test-tool/PackageManagementTest.dar"
+    com.daml.ledger.test_common.TestDars.fileNames("package_management")
 
   private def loadTestPackage()(implicit ec: ExecutionContext): Future[ByteString] = {
     val testPackage = Future {
-      val in = getClass.getResourceAsStream(testPackageResourcePath)
+      val in = getClass.getClassLoader.getResourceAsStream(testPackageResourcePath)
```
Comment (cocreature) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/PackageManagementServiceIT.scala:23]: why do we need to throw a `getClassLoader` in between now?

File: ledger/test-common/BUILD.bazel [hunk 8]
```diff
@@ -14,24 +14,28 @@ load(
...
-    (lf_stable_version, ""),
-    (lf_latest_version, "-latest"),
-    (lf_dev_version, "-dev"),
-] + [
-    (v, "-preview")
-    for v in lf_preview_version
+[
+    [
+        alias(
+            name = "%s-%s%s" % (target_prefix, name, target_suffix),
```
Comment (cocreature) [inline ledger/test-common/BUILD.bazel:24]: it’s hard to see what is actually being aliased here quickly. I think a comment for future readers would be helpful with an example.

**Review by cocreature — APPROVED** [review]
Thanks!

**Review by aherrmann-da — APPROVED** [review]

**Review by gerolf-da — APPROVED** [review]
Thanks @remyhaemmerle-da

File: ledger/test-common/BUILD.bazel [hunk 9]
```diff
@@ -89,21 +107,51 @@ da_scala_library(
...
 ]
 
 [
     [
+        genrule(
+            name = "dar-files-lookup-%s.scala" % target_suffix,
+            outs = ["Dars-%s.scala" % target_suffix],
+            cmd = """
+cat > $@ <<EOF
+package com.daml.ledger.test_common
```
Comment (miklos-da) [inline ledger/test-common/BUILD.bazel:122]: Please follow the [Java package naming conventions](https://google.github.io/styleguide/javaguide.html#s5.2.1-package-names), i.e., don't use underscores. If you don't want to end up with `testcommon`, then please create a package `com.daml.ledger.test.common` or similar.

**Review by miklos-da — CHANGES_REQUESTED** [review]

**Review by miklos-da — APPROVED** [review]

**Pull request merged**